src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -165,4 +167,123 @@ public void testAbstractInterfaceConcept() {
+        student.setRollNo(rollNo);
+        Student.setSchoolName(schoolName);
+
+        student2.setSchoolName("SVP");
+
+        softAssert.assertThat(student.getName()).as("Name is not matching").isEqualTo(name);
+
+        softAssert.assertThat(student.getLastName()).as("Lastname is not matching").isEqualTo(lastName);
+
+        softAssert.assertThat(student.getRollNo()).as("Roll number doesn't match").isEqualTo(rollNo);
+
+        softAssert.assertThat(Student.getSchoolName()).as("School name doesn't match").isEqualTo("SVP");
+
+        softAssert.assertThat(student2.getSchoolName()).as("School name doesn't match").isEqualTo("SVP");
+    }
+
+    @Test
+    public void testMiniProject() {
+        com.training.assignments.miniproject.Employee originalEmployee =
+                new com.training.assignments.miniproject.Employee();
+
+        com.training.assignments.miniproject.Employee cloneEmployee =
+                new com.training.assignments.miniproject.Employee();
+
+        List<com.training.assignments.miniproject.Employee> createEmp = originalEmployee.getCreatedEmployee();
+        List<com.training.assignments.miniproject.Employee> cloneEmp = cloneEmployee.getExistingEmployee();
+
+        Iterator<com.training.assignments.miniproject.Employee> createdEmployeeIterator = createEmp.iterator();
+        Iterator<com.training.assignments.miniproject.Employee> cloneEmployeeIterator = cloneEmp.iterator();
+
+        while (createdEmployeeIterator.hasNext()) {
+            com.training.assignments.miniproject.Employee empDetails = createdEmployeeIterator.next();
+            com.training.assignments.miniproject.Employee cloneEmpDetails = cloneEmployeeIterator.next();
+
+            softAssert.assertThat(empDetails.getName())
+                    .as("Name's inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getName());
+
+            softAssert.assertThat(empDetails.getAddress())
+                    .as("Address inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getAddress());
+
+            softAssert.assertThat(empDetails.getId())
+                    .as("Id's inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getId());
+
+            softAssert.assertThat(empDetails.getSalary())
+                    .as("Salary inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getSalary());
+
+        }
+
+        List<com.training.assignments.miniproject.Employee> deletedEmployee = originalEmployee.getDeletedEmployee(1);
+        List<com.training.assignments.miniproject.Employee> cloneDeletedEmp = cloneEmployee.getCloneDeletedEmployee(1);
+
+        Iterator<com.training.assignments.miniproject.Employee> deletedEmployeeIterator = deletedEmployee.iterator();
+        Iterator<com.training.assignments.miniproject.Employee> cloneDeletedEmployeeIterator = cloneDeletedEmp.iterator();
```

Comment:
Dont create list deletedEmployee , cloneDeletedEmp , pass function call direclty in .iterator()
Iterator<com.training.assignments.miniproject.Employee> deletedEmployeeIterator = originalEmployee.getDeletedEmployee(1).iterator();